workspaces/liferay-ticket-workspace/client-extensions/liferay-ticket-custom-element/src/components/TicketCard.tsx

```diff
@@ -0,0 +1,165 @@
+import {useContext, useState} from 'react';
+import {QueryClient, useMutation} from 'react-query';
+
+import {QueryClientContext} from '../context';
+import {Liferay} from '../services/liferay';
+import {assignTicketToMe} from '../services/tickets';
+import {Ticket} from '../types';
+
+const TicketCard: React.FC<{ticket: Ticket}> = ({ticket}) => {
+	const [isLoading, setIsLoading] = useState(false);
+	const [isPanelExpanded, setIsPanelExpanded] = useState(false);
+
+	const queryClient: QueryClient = useContext(QueryClientContext);
+
+	const {
+		attributes,
+		isDragging,
+		listeners,
+		setNodeRef,
+		transform,
+	} = useDraggable({data: ticket, id: ticket.id + '_draggable'});
+
+	const style: React.CSSProperties = {
+		position: 'relative',
+		transform: CSS.Translate.toString(transform),
+		zIndex: isDragging ? 150 : 1,
+	};
+
+	const assignToMeMutation = useMutation({
+		mutationFn: async (ticket: Ticket) => {
+			setIsLoading(true);
+
+			return await assignTicketToMe(ticket);
+		},
+		onError: (error: Error) => {
+			setIsLoading(false);
+
+			Liferay.Util.openToast({
+				message: error.message,
+				title: 'Request Failed',
+				type: 'danger',
+			});
+		},
+		onSuccess: () => {
+			setIsLoading(false);
+
+			queryClient.invalidateQueries();
+
+			Liferay.Util.openToast({
+				message: 'The ticket was successfully assigned to you!',
+				type: 'success',
+			});
+		},
+	});
+
+	return (
+		<div
+			{...attributes}
+			className={`border border-neutral-2 card mb-4 py-2 shadow-none ${
+				isDragging ? 'bg-brand-primary-lighten-6' : 'bg-neutral-0'
+			}`}
+			ref={setNodeRef}
+			style={style}
+		>
+			<ClayLayout.ContentRow padded>
+				<ClayLayout.ContentCol
+					className="justify-content-center p-2"
+					shrink
+				>
+					<ClayIcon
+						{...listeners}
+						style={{cursor: 'grab'}}
+						symbol="drag"
+					></ClayIcon>
+				</ClayLayout.ContentCol>
+				<ClayLayout.ContentCol shrink>
+					<ClayPanel
+						className="border-0 m-0 p-0"
+						collapsable
+						displayTitle={
+							<ClayPanel.Title>
+								<div
+									className={`text-neutral-9 font-weight-bold ${
+										!isPanelExpanded
+											? 'text-truncate text-paragraph-sm'
+											: 'text-paragraph'
+									}`}
+								>
+									{ticket.subject}
+								</div>
+								<div className="font-weight-normal mt-3 text-neutral-8 text-paragraph-xs">
+									{ticket.assignee ? (
+										<>
+											<ClayIcon
+												className="mr-1 rounded-circle"
+												symbol="user"
+											></ClayIcon>
+
+											<div className="d-inline">
+												{`${ticket.assignee?.givenName} ${ticket.assignee?.familyName}`}
+											</div>
+										</>
+									) : (
+										<div className="ml-2">
+											Not assigned.
+										</div>
+									)}
+								</div>
+							</ClayPanel.Title>
+						}
+						displayType="secondary"
+						expanded={isPanelExpanded}
+						onExpandedChange={(isExpanded: any) =>
```

Comment:
We should always avoid `any` type whenever possible. This looks like an easy one to avoid... either boolean or boolean | undefined. Check for all other usages of `any` type.